Unable to build on Ubuntu 18.04

Errormessage:

```
/usr/src/ts3init-latest/src/ts3init_match.c: In function ‘calculate_cookie’:
/usr/src/ts3init-latest/src/ts3init_match.c:144:16: error: ‘const struct xt_action_param’ has no member named ‘family’
     switch (par->family)
                ^~
/usr/src/ts3init-latest/src/ts3init_match.c:177:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
```

--- a/src/Makefile.xtables
+++ b/src/Makefile.xtables
@@ -6,7 +6,11 @@ clean:
 	rm -f $(LIBS)
 
 install:
-	install -g root -o root -m 644 $(LIBS) /lib/xtables/
+	if [ -d /usr/lib/x86_64-linux-gnu/xtables ]; then \
+      install -g root -o root -m 644 $(LIBS) /usr/lib/x86_64-linux-gnu/xtables; \
+    else \
+      install -g root -o root -m 644 $(LIBS) /lib/xtables/ ; \
+    fi
 
 lib%.so: lib%.o
 	gcc -shared -fPIC -o $@ $^;

--- a/src/Makefile.xtables
+++ b/src/Makefile.xtables
@@ -13,8 +13,8 @@ install:
 	elif [ -d /usr/lib/i386-linux-gnu/xtables ]; then \
 	  install -g root -o root -m 644 $(LIBS) /usr/lib/i386-linux-gnu/xtables; \
 	else \
-	  echo "Unable to find xtables modules path!"
-	  exit 1
+	  echo "Unable to find xtables modules path!"; \
+	  exit 1; \
 	fi
 
 lib%.so: lib%.o

--- a/src/Makefile.xtables
+++ b/src/Makefile.xtables
@@ -6,11 +6,16 @@ clean:
 	rm -f $(LIBS)
 
 install:
-	if [ -d /usr/lib/x86_64-linux-gnu/xtables ]; then \
-      install -g root -o root -m 644 $(LIBS) /usr/lib/x86_64-linux-gnu/xtables; \
-    else \
-      install -g root -o root -m 644 $(LIBS) /lib/xtables/ ; \
-    fi
+	if [ -d /lib/xtables ]; then \
+	  install -g root -o root -m 644 $(LIBS) /lib/xtables/ ; \
+	elif [ -d /usr/lib/x86_64-linux-gnu/xtables ]; then \
+	  install -g root -o root -m 644 $(LIBS) /usr/lib/x86_64-linux-gnu/xtables; \
+	elif [ -d /usr/lib/i386-linux-gnu/xtables ]; then \
+	  install -g root -o root -m 644 $(LIBS) /usr/lib/i386-linux-gnu/xtables; \
+	else \
+	  echo "Unable to find xtables modules path!"
+	  exit 1
+	fi
 
 lib%.so: lib%.o
 	gcc -shared -fPIC -o $@ $^;

--- a/src/compat_xtables.h
+++ b/src/compat_xtables.h
@@ -90,7 +90,7 @@ static inline void proc_remove(struct proc_dir_entry *de)
 static inline struct net *par_net(const struct xt_action_param *par)
 {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0)
-    return par->net;
+    return xt_net(par);
 #else
     return dev_net((par->in != NULL) ? par->in : par->out);
 #endif

--- a/src/compat_xtables.h
+++ b/src/compat_xtables.h
@@ -89,8 +89,10 @@ static inline void proc_remove(struct proc_dir_entry *de)
 
 static inline struct net *par_net(const struct xt_action_param *par)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
     return xt_net(par);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0)
+    return par->net;
 #else
     return dev_net((par->in != NULL) ? par->in : par->out);
 #endif

--- a/src/ts3init_match.c
+++ b/src/ts3init_match.c
@@ -141,7 +141,11 @@ static inline __u8* get_payload(const struct sk_buff *skb, const struct xt_actio
 static int calculate_cookie(const struct sk_buff *skb, const struct xt_action_param *par, 
                        struct udphdr *udp, __u64 k0, __u64 k1, __u64* out)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
     switch (xt_family(par))
+#else
+    switch (par->family)
+#endif
     {
     case NFPROTO_IPV4:
         {

--- a/src/ts3init_match.c
+++ b/src/ts3init_match.c
@@ -141,7 +141,7 @@ static inline __u8* get_payload(const struct sk_buff *skb, const struct xt_actio
 static int calculate_cookie(const struct sk_buff *skb, const struct xt_action_param *par, 
                        struct udphdr *udp, __u64 k0, __u64 k1, __u64* out)
 {
-    switch (par->family)
+    switch (xt_family(par))
     {
     case NFPROTO_IPV4:
         {

--- a/src/ts3init_match.c
+++ b/src/ts3init_match.c
@@ -13,6 +13,7 @@
  */
 
 #include <linux/kernel.h>
+#include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/netfilter/x_tables.h>

--- a/src/ts3init_target.c
+++ b/src/ts3init_target.c
@@ -47,7 +47,12 @@ ts3init_send_ipv6_reply(struct sk_buff *oldskb, const struct xt_action_param *pa
     struct udphdr *udp;
     struct flowi6 fl;
     struct dst_entry *dst = NULL;
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
     struct net *net = dev_net((xt_in(par) != NULL) ? xt_in(par) : xt_out(par));
+#else
+    struct net *net = dev_net((par->in != NULL) ? par->in : par->out);
+#endif
 
     skb = alloc_skb(LL_MAX_HEADER + sizeof(*ip) +
              sizeof(*udp) + payload_size, GFP_ATOMIC);

--- a/src/ts3init_target.c
+++ b/src/ts3init_target.c
@@ -13,6 +13,7 @@
  */
 
 #include <linux/kernel.h>
+#include <linux/version.h>
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/skbuff.h>

--- a/src/ts3init_target.c
+++ b/src/ts3init_target.c
@@ -47,7 +47,7 @@ ts3init_send_ipv6_reply(struct sk_buff *oldskb, const struct xt_action_param *pa
     struct udphdr *udp;
     struct flowi6 fl;
     struct dst_entry *dst = NULL;
-    struct net *net = dev_net((par->in != NULL) ? par->in : par->out);
+    struct net *net = dev_net((xt_in(par) != NULL) ? xt_in(par) : xt_out(par));
 
     skb = alloc_skb(LL_MAX_HEADER + sizeof(*ip) +
              sizeof(*udp) + payload_size, GFP_ATOMIC);

